### ISSUE TITLE
Jacoco coverage report for Android Tests doesn't work after Mamification on Android

### ISSUE THREAD

**Issue by shubhamgarg1** [msintuneappsdk/ms-intune-app-sdk-android#145]
**Describe the bug:**
We are trying to generate Jacoco Code Coverage report in our test module but some classes where the code is replaced by their intune counterparts don't work correctly for e.g. onCreate method in activities is shown as not covered despite being covered in tests. It works correctly if we don't use the intune plugin or if we manually specify intune transform directory using a logic like below in the jacoco coverage task:

```
def files = []
def i = 0
while (i < 10000) {
    files.add(new File("${transformFolder.path}/${i.toString()}/"))
     i++
}
getClassDirectories().setFrom(files)
```

**To Reproduce**
1. Clone this repository https://github.com/shubhamgarg1/IntuneCodeCoverageNotWorking/commits/main
2. Run the commands `./gradlew -is :testOnlyModule:connectedDebugAndroidTest :testOnlyModule:createAndroidTestCoverage`
3. Check the code coverage report inside testOnlyModule/build/reports/jacocoCoverageReport/html/index.html

**Expected behavior:**
The report should include methods like onCreate, onResume in the report but it is shown as not covered.

**Screenshots and logs:**
The screenshot after using Intune:
<img width="992" alt="Screenshot 2023-01-07 at 11 43 59 AM" src="https://user-images.githubusercontent.com/14014771/211134135-dc852f83-bd67-4f14-b6c8-1f81b0b7976e.png">

Correct Screenshot without intune:
<img width="1088" alt="Screenshot 2023-01-07 at 11 46 49 AM" src="https://user-images.githubusercontent.com/14014771/211134219-f0d7c217-755f-427b-b77f-f52cd60fa19e.png">

**Smartphone (please complete the following information):**
 - Device: All
 - OS: All

**Intune App SDK for Android (please complete the following information):**
- What version of the Intune SDK are you using? Are you using the latest version? Yes
- What platform is your app based in (Java, Xamarin based, Cordova, etc)? Java
- For errors during build, does the app build without Intune SDK integration? No
- For errors post build, does the app launch without being Intune SDK integrated? No
- Who is the customer? Adobe
- Do you see a trend with it only being reproduced on a specific device? No

We would like to know how should we be using intune together with jacoco to generate correct Android Test reports?

**Comment by bannus** [comment]
Hi @shubhamgarg1, thanks for the report. Jacoco coverage after Mamification isn't currently supported. Is it possible for you to run your coverage reporting on a non-Mamified flavor of your app?

##### (internal tracking #16821065)

**Comment by shubhamgarg1** [comment]
HI @bannus , yes, we can run our tests on a non-mamified version but we were looking to have the testing environment as close as possible to the release build. As in our release build, we have Mamification as on, but in our testing, we would not be able to discover any issues if they happen because of the plugin.

**Comment by bannus** [comment]
I've confirmed that Jacoco coverage works with the provided sample app with the `-preview` version of the MAM SDK build plugin, [available in version 9.4.0](https://github.com/msintuneappsdk/ms-intune-app-sdk-android/blob/master/CHANGELOG.md#version-940) and above. (Note that the preview version will become the default for apps consuming AGP 8+).
